rocketpy/Rocket.py

```diff
@@ -537,56 +528,133 @@ def addFins(
-        Lf = np.sqrt((Cr / 2 - Ct / 2) ** 2 + s**2)
+        # Checking the dictionary
+        if "rootChord" not in finParameters.keys():
+            raise Exception(
+                "rootChord dictionary key is missing from finParameters")
+        if "span" not in finParameters.keys():
+            raise Exception(
+                "span dictionary key is missing from finParameters")
+
+        if type == "trapezoidal":
+            if "tipChord" not in finParameters.keys():
+                raise Exception(
+                    "tipChord dictionary key is missing from finParameters")
+
+        # Retrieves similar parameters
+        Cr = finParameters["rootChord"]
+        s = finParameters["span"]
         radius = self.radius if radius == 0 else radius
+        cantAngleRad = np.radians(cantAngle)
         d = 2 * radius
+
+        # Type verification
+        if type == "trapezoidal":
+
+            # Retrieve parameters for calculations
+            Ct = finParameters["tipChord"]
+            Yr = Cr + Ct
+
+            Af = Yr * s / 2  # fin area
+            gamac = np.arctan((Cr - Ct) / (2 * s))  # mid chord angle
+            Ymac = (
+                (s / 3) * (Cr + 2 * Ct) / Yr
+            )  # span wise position of fin's mean aerodynamic chord
+            Lf = np.sqrt((Cr / 2 - Ct / 2) ** 2 + s**2)  #
+
+            rollGeometricalConstant = (
+                (Cr + 3 * Ct) * s**3
+                + 4 * (Cr + 2 * Ct) * radius * s**2
+                + 6 * (Cr + Ct) * s * radius**2
+            ) / 12
+
+            # Calculate cp position relative to cm
+            if distanceToCM < 0:
+                cpz = distanceToCM - (
+                    ((Cr - Ct) / 3) * ((Cr + 2 * Ct) / (Cr + Ct))
+                    + (1 / 6) * (Cr + Ct - Cr * Ct / (Cr + Ct))
+                )
+            else:
+                cpz = distanceToCM + (
+                    ((Cr - Ct) / 3) * ((Cr + 2 * Ct) / (Cr + Ct))
+                    + (1 / 6) * (Cr + Ct - Cr * Ct / (Cr + Ct))
+                )
+
+        elif type == "elliptical":
+
+            # Retrieve parameters for calculations
+            Af = (np.pi * Cr / 2 * s) / 2
+            gamac = 0
+            Ymac = s / (3 * np.pi) * np.sqrt(9 * np.pi**2 - 64)
+            rollGeometricalConstant = (
+                Cr
+                * s
+                * (3 * np.pi * s**2 + 32 * radius * s + 12 * np.pi * radius**2)
+                / 48
+            )
+
+            # Calculate cp position relative to cm
+            if distanceToCM < 0:
+                cpz = distanceToCM - (0.288 * Cr)
+            else:
+                cpz = distanceToCM + (0.288 * Cr)
+
+        else:
+            raise Exception("Invalid fin shape")
+
+        # Retrieves similar parameters
         Aref = np.pi * radius**2
-        AR = 2 * s**2 / Af  # Barrowman's convention for fin's aspect ratio
-        cantAngleRad = np.radians(cantAngle)
-        trapezoidalConstant = (
```

Comment:
The `trapezoidalConstant` here was left behind, this will start a problem on the line 760 when you try to calculate `cldOmega` by using `trapezoidalConstant` value.

After a deeper look I realized this is related and solve the previous commentarie since it looks like `trapezoidalConstant` was renamed to `rollGeometricalConstant`